src/asset_folder_importer/premiere_get_referenced_media/processor.py

```diff
@@ -259,6 +253,9 @@ def process_premiere_project(filepath, raven_client, vs_pathmap=None, db=None, c
                 lg.info("File %s with id %s is already linked to project %s" % (filepath.encode('utf-8'), e.fileid, e.vsprojectid))
             continue
 
+    if invalid_media_paths != []:
+        vsproject.set_metadata({'gnm_project_invalid_media_paths': invalid_media_paths}, mode="add")
```

Comment:
please could you separate this into its own method and:

- check the current metadata value (if `vsproject` is populated already don't `populate` another object, just use `vsproject.get()`. You may have to use a specific flag to allow `get` to return lists)
- only run the update if the value we are going to set is different to the existing value
- an empty value is valid, so long as the previous value was not empty; this would mean that the user has fixed their project and we should make the warning go away
- give it its own test scenario, which should test for the cases of previous value empty/new value empty, previous value empty/new value has list, previous value has list/new value empty, previous value has list/new value has same list, previous value has list/new value has updated list with items added, previous value has list/new value has updated list with items removed